### ISSUE TITLE
Custom domain configuration for storage accounts not working when using subdomain

### ISSUE THREAD

**Issue by rgiese** [Azure/azure-sdk-for-python#4013]
The following command does not correctly leverage the `asverify` subdomain for verification: `az storage account update --name ... --resource-group ... --custom-domain example.com --use-subdomain true`

Running the command with `--debug` shows a request of `"properties": {"customDomain": {"name": "True"}, "encryption"...`, so the `useSubDomain` parameter has somehow been dropped.

Looking at the model in `azure-mgmt-storage/azure/mgmt/storage/v2018_02_01/models/custom_domain.py` shows it trying to create a request snippet of `'key': 'useSubDomain'`; however, if you inspect the network requests issued by the Azure Portal (where this does work), it instead sends a key of `"useSubDomainName"` - not the addition of *Name*.

I suspect that somewhere in here the Azure REST service made a breaking change, renaming `useSubDomain` to `useSubDomainName` and either the REST API needs to be poked in the eye or all the client SDKs need to be versioned.

**Comment by lmazuel** [comment]
Hi @rgiese , sorry the late answer, and thank you for the precise investigation!

@blueww @williexu what do you think?

**Comment by blueww** [comment]
@lmazuel , @rgiese 

I just tested , looks  "useSubDomain" still can be set in rest API, but set it seems not take any effect in the returned account properties.
I will contact SRP team to check what happened.
 
See the request as following.

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PATCH

Absolute Uri:
https://management.azure.com/subscriptions/c***/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***?api-version=2018-07-01

Headers:
x-ms-client-request-id        : 4f1cd02a-1d5a-408d-b39f-63cbf909149d
accept-language               : en-US

Body:
{
  "properties": {
    "customDomain": {
      "name": "dmtestsrp2.sn3.azure-test.net",
      "useSubDomain": true
    }
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Pragma                        : no-cache
x-ms-request-id               : 6d3fb6d8-b641-4e5d-8329-9a78c5088f50
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : 30e7b654-1d7f-4fbe-a01f-a8383c84cb06
x-ms-routing-request-id       : SOUTHEASTASIA:20181217T020818Z:30e7b654-1d7f-4fbe-a01f-a8383c84cb06
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Mon, 17 Dec 2018 02:08:17 GMT
Server                        : Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0

Body:
{
  "sku": {
    "name": "Standard_GRS",
    "tier": "Standard"
  },
  "kind": "Storage",
  "id": "/subscriptions/***/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***",
  "name": "***",
  "type": "Microsoft.Storage/storageAccounts",
  "location": "westus",
  "tags": {},
  "properties": {
    "networkAcls": {
      "bypass": "AzureServices",
      "virtualNetworkRules": [],
      "ipRules": [],
      "defaultAction": "Allow"
    },
    "supportsHttpsTrafficOnly": false,
    "encryption": {
      "services": {
        "file": {
          "enabled": true,
          "lastEnabledTime": "2018-12-17T02:07:13.2746515Z"
        },
        "blob": {
          "enabled": true,
          "lastEnabledTime": "2018-12-17T02:07:13.2746515Z"
        }
      },
      "keySource": "Microsoft.Storage"
    },
    "provisioningState": "Succeeded",
    "creationTime": "2018-12-17T02:07:13.1496219Z",
    "primaryEndpoints": {
      "blob": "https://***.blob.core.windows.net/",
      "queue": "https://***.queue.core.windows.net/",
      "table": "https://***.table.core.windows.net/",
      "file": "https://***.file.core.windows.net/"
    },
    "primaryLocation": "westus",
    "statusOfPrimary": "available",
    "secondaryLocation": "eastus",
    "statusOfSecondary": "available",
    "customDomain": {
      "name": "dmtestsrp2.sn3.azure-test.net"
    }
  }
}
```

**Comment by lmazuel** [comment]
Hi @rgiese 
There was a double layers of issue here:
- CLI was incorrectly using "name" to pass the --use-subdomain (see https://github.com/Azure/azure-cli/pull/8079)
- useSubDomain should likely be called useSubDomainName

We're working on the correct fix right now. Thanks!

**Comment by lmazuel** [comment]
Fixed in 3.1.1 https://pypi.org/project/azure-mgmt-storage/3.1.1/

@rgiese let us know if you still encounter an issue related to this. Thank you!
